### DISCUSSION TITLE
Why is my page load over 1.5 second with so less data?

### DISCUSSION THREAD

**Discussion by SelfDevTV** [vercel/next.js#20744] [Help]
Hi guys,

I'm building a reddit clone right now. I'm fetching a list of a subreddit posts in `getServerSideProps`. This function makes an api call to my prisma database (hosted on Amazon AWS - PostgreSQL). 

Here is the working live example (just select a subreddit in the dropdown). Next.js has some posts, learnprogramming hasnt: https://reddit-clone-teal.vercel.app/

**Here is the api:**

```
const handler = async (req: NextApiRequest, res: NextApiResponse) => {
  try {
    const sub = await prisma.subreddit.findUnique({
      where: { name: String(req.query.name) },
      include: {
        posts: {
          include: { subreddit: true, user: true },
        },
        joinedUsers: true
      },
    });

    console.log("here is the sub", sub);

    if (!sub) {
      return res.status(500).json({ error: "No sub was found with this name" });
    }

    res.json(sub);
  } catch (e) {
    res.json(e);
  }
};
```

So nothing too crazy right (it's 3 posts total)? Or am I including too much in the call?
Anyway when I transition to a subreddit and this function runs (there are currently 3 dummy posts in the db) it takes sometimes 1.5 - 2 seconds after the page has loaded in.
With such little data this should be much less loading time am I right?

I'm using `swr` and here is how a subreddit page looks like:

**First the `getServerSideProps` function:**

```
export async function getServerSideProps(ctx) {
  const url = `${process.env.NEXTAUTH_URL}/api/subreddit/findSubreddit?name=${ctx.query.sub}`;
  const fullSub = await fetchData(url);

  return {
    props: {
      fullSub,
    },
  };
}
```

**And then on the page itself:**

```
const { data: fullSub, error } = useSWR(
    `/api/subreddit/findSubreddit?name=${sub}`,
    fetchData,
    {
      initialData: props.fullSub,
    }
  );
```

It also kinda loads it everytime (going from one subreddit to another), although I'm caching it with `swr`. I think I'm doing something wrong. It's kinda faster then the first transition from subreddit A to subreddit B but shouldn't it be instant when nothing has changed? Isn't that the advantage of using `swr` or am I missing something again? 😄 

Can you figure it out? Any ideas why this might long load and not cache?

Full link to my repo: https://github.com/SelfDevTV/reddit-clone

Thank you so much!
_**PS: Next.js rocks!**_

**Comment by BramDecuypere** [comment]
I'd advise you to upload your sourcemaps as well to check what is actually going on! And check what is being loaded and what makes it slow! Hope that helps.
- Have a look at bundle-wizard
- webpagetest.org
- Chrome debugger

And you'll definitely see what's really going on. 

For me this looks quite fast at this point. At least without logging in.
